app/controllers/api/v0/versions_controller.rb

```diff
@@ -25,6 +27,41 @@ def show
     }
   end
 
+  def raw
+
+    @version ||= version_collection.find(params[:id])
+
+    def self.public_hosts=(hosts)
+      hosts = [] if hosts.nil?
+      hosts = hosts.split(' ') if hosts.is_a?(String)
+      unless hosts.is_a?(Enumerable) && hosts.all? {|host| host.is_a?(String)}
+        raise StandardError, 'Public hosts must be a string or enumerable of strings'
+      end
+      @public_hosts = hosts
+    end
+
+    if ENV['PUBLIC_ARCHIVE_HOSTS']
+      self.public_hosts = ENV['PUBLIC_ARCHIVE_HOSTS']
```

Comment:
It *seems* to me like this concept of “public archive hosts” could be a superset of what’s in `ALLOWED_ARCHIVE_HOSTS`. If the `uri` points to a place that we didn’t archive locally because it’s in an accessible, public location, that might be a good indicator that we should redirect to it. That said, I can see how those are technically separate concepts, and there could be some future situation where this wouldn’t be a strict superset of that. I guess I’m not really sure if that’s actually a useful distinction in practice.

Also, I *generally* prefer keeping things like environment variables confined to the `config` directory (usually by setting up things in an initializer), that way it’s slightly more possible to keep track of what env vars are used where.

Reply:
Makes sense. I setup the public archive check method to test for public or allowed hosts and made the environment variable optional. I also moved the environment variable handling into the archiver initializer since I moved the related checking/fetching methods into archiver as well.

Reply:
this has been dispensed with